core/src/subgraph/instance_manager.rs

```diff
@@ -331,6 +331,19 @@ impl<S: SubgraphStore> SubgraphInstanceManager<S> {
             })
             .collect();
 
+        // We can set `max_end_block` to the maximum of `end_blocks` and stop the subgraph
+        // at that point only when there are no dynamic data sources present and offchain
+        // data sources exist. This is because:
+        // - Dynamic data sources do not have a defined `end_block`, so we can't determine
+        //   when to stop processing them.
+        // - Offchain data sources might still require processing beyond the end block of
+        //   onchain data sources, necessitating the continuation of the subgraph.
```

Comment:
```suggestion
        // We can set `max_end_block` to the maximum of `end_blocks` and stop the subgraph
        // at that point only when there are no dynamic data sources present and offchain
        // data sources exist. This is because:
        // - Dynamic data sources do not have a defined `end_block`, so we can't determine
        //   when to stop processing them.
        // - Offchain data sources might still require processing beyond the end block of
        //   onchain data sources, necessitating the continuation of the subgraph.
```